doc/ale-c.txt

```diff
@@ -22,6 +53,104 @@ g:ale_c_clang_options                                   *g:ale_c_clang_options*
 
 
 ===============================================================================
+clangtidy                                                   *ale-c-clangtidy*
+
+`clang-tidy` will be run only when files are saved to disk, so that
+`compile_commands.json` files can be used. It is recommended to use this
+linter in combination with `compile_commands.json` files.
+Therefore, `clang-tidy` linter reads the options |g:ale_c_build_dir| and
+|g:ale_c_build_dir_names|. Also, setting |g:ale_c_build_dir| actually
+overrides |g:ale_c_build_dir_names|.
+
+
+g:ale_c_clangtidy_checks                             *g:ale_c_clangtidy_checks*
+                                                     *b:ale_c_clangtidy_checks*
+  Type: |List|
+  Default: `['*']`
+
+  The checks to enable for clang-tidy with the `-checks` argument.
+
+  All options will be joined with commas, and escaped appropriately for
+  the shell. The `-checks` flag can be removed entirely by setting this
+  option to an empty List.
+
+  Not all of clangtidy checks are applicable for C. You should consult the
+  clang documentation for an up-to-date list of compatible checks.
+
+  Here's an (incomplete) list of C-compatible checks, for clang-tidy version 6:
+    bugprone-suspicious-memset-usage
+    cert-env33-c
+    cert-err34-c
+    cert-flp30-c
+    google-runtime-int
+    llvm-header-guard
+    llvm-include-order
+    misc-argument-comment
+    misc-assert-side-effect
+    misc-bool-pointer-implicit-conversion
+    misc-definitions-in-headers
+    misc-incorrect-roundings
+    misc-macro-parentheses
+    misc-macro-repeated-side-effects
+    misc-misplaced-const
+    misc-misplaced-widening-cast
+    misc-multiple-statement-macro
+    misc-non-copyable-objects
+    misc-redundant-expression
+    misc-sizeof-expression
+    misc-static-assert
+    misc-string-literal-with-embedded-nul
+    misc-suspicious-enum-usage
+    misc-suspicious-missing-comma
+    misc-suspicious-semicolon
+    misc-suspicious-string-compare
+    misc-swapped-arguments
+    modernize-redundant-void-arg
+    modernize-use-bool-literals
+    performance-type-promotion-in-math-fn
+    readability-braces-around-statements
+    readability-else-after-return
+    readability-function-size
+    readability-identifier-naming
+    readability-implicit-bool-cast
+    readability-inconsistent-declaration-parameter-name
+    readability-misleading-indentation
+    readability-misplaced-array-index
+    readability-named-parameter
+    readability-non-const-parameter
+    readability-redundant-control-flow
+    readability-redundant-declaration
+    readability-redundant-function-ptr-dereference
+    readability-simplify-boolean-expr
+
+
+g:ale_c_clangtidy_executable                 *g:ale_c_clangtidy_executable*
```

Comment:
Nah, I actually prefer having many commits for pull requests, so I can see the difference before and after reviews. Then I just squash the pull request into one commit after that via GitHub's button for `--ff --squash`.